### ISSUE TITLE
Empty file saved as a directory

### ISSUE THREAD

**Issue by BioEvo** [miurahr/py7zr#305]
**Describe the bug**
when a empty file is write to a 7z archive, it will become a directory.

**To Reproduce**

1. Create a empyt file

> touch x

2. Run following code with python3.

>from py7zr import SevenZipFile as p7z
>archive = p7z("test.7z", 'w', password='123')
>archive.set_encrypted_header(True)
>archive.write("x", "y")
>archive.close()

3. test.7z is produced

**Expected behavior**
There should be an empty file "y" in archive "test.7z", but we got a directory "y" in "test.7z" instead.

**Environment (please complete the following information):**
 - OS: CentOS
 - Python 3.9.1
 - py7zr version: 0.13.0

**Comment by miurahr** [comment]
```
@pytest.mark.files
def test_encrypt_file_7(tmp_path):
    tmp_path.joinpath('src').mkdir()
    with tmp_path.joinpath('src', 'x').open(mode='wb') as f:
        f.write(b'')
    archive = py7zr.SevenZipFile(tmp_path.joinpath("target.7z"), 'w', password='123')
    archive.set_encrypted_header(True)
    archive.write(tmp_path.joinpath('src', 'x'), 'y')
    archive.close()
    #
    with py7zr.SevenZipFile(tmp_path.joinpath("target.7z"), 'r', password='123') as arc:
        arc.extractall(path=tmp_path / "tgt")
    #
    assert tmp_path.joinpath('tgt', 'y').is_file()

```

This test case passed and extracted FILE not Directory.
Examining produced archive, there is no DIRECTORY flag on file y.

The problem seems an issue on `p7zip` side that cannot treat a file of emptystream that is not flagged as directory.

**Comment by miurahr** [comment]
Behavior changed at v0.12.0 (#288) to fix the issue(#287) when adding empty data file in append mode.

The commit: https://github.com/miurahr/py7zr/commit/ba235f27024b164ad2215f60a64848e2508b7bad
